x-pack/plugins/uptime/server/uptime_server.ts

```diff
@@ -5,21 +5,57 @@
  * 2.0.
  */
 
-import { UMServerLibs } from './lib/lib';
+import { Logger } from 'kibana/server';
+import { createUptimeESClient, UMServerLibs } from './lib/lib';
 import { createRouteWithAuth, restApiRoutes, uptimeRouteWrapper } from './rest_api';
 import { UptimeCoreSetup, UptimeCorePlugins } from './lib/adapters';
 import { uptimeAlertTypeFactories } from './lib/alerts';
+import { createLifecycleRuleTypeFactory, RuleDataClient } from '../../rule_registry/server';
+import { savedObjectsAdapter } from './lib/saved_objects';
 
 export const initUptimeServer = (
   server: UptimeCoreSetup,
   libs: UMServerLibs,
-  plugins: UptimeCorePlugins
+  plugins: UptimeCorePlugins,
+  ruleDataClient: RuleDataClient,
+  logger: Logger
 ) => {
   restApiRoutes.forEach((route) =>
     libs.framework.registerRoute(uptimeRouteWrapper(createRouteWithAuth(libs, route)))
   );
 
-  uptimeAlertTypeFactories.forEach((alertTypeFactory) =>
-    plugins.alerting.registerType(alertTypeFactory(server, libs, plugins))
-  );
+  uptimeAlertTypeFactories.forEach((alertTypeFactory) => {
+    const alertType = alertTypeFactory(server, libs, plugins);
+    const createLifecycleRuleType = createLifecycleRuleTypeFactory({
+      ruleDataClient,
+      logger,
+    });
+
+    return plugins.alerting.registerType(
+      createLifecycleRuleType({
+        ...alertType,
+        producer: 'uptime',
+        executor: async ({ services, ...rest }) => {
+          const { scopedClusterClient, savedObjectsClient } = services;
+          const dynamicSettings = await savedObjectsAdapter.getUptimeDynamicSettings(
+            savedObjectsClient
+          );
```

Comment:
i wonder if we need this explicitely here since `uptimeEsClient` should take care of fetcing this when search call happens.